go/enclave/events/subscription_manager.go

```diff
@@ -116,9 +116,16 @@ func (s *SubscriptionManager) FilteredLogs(logs []*types.Log, rollupHash common.
 	return filteredLogs
 }
 
-// FilteredSubscribedLogs filters out irrelevant logs, those that are not subscribed to, and those a subscription has
-// seen before, and organises them by their subscribing ID.
-func (s *SubscriptionManager) FilteredSubscribedLogs(logs []*types.Log, rollupHash common.L2RootHash) map[gethrpc.ID][]*types.Log {
+// EncryptFilteredLogs filters the logs based on the current subscriptions, then encrypts them the appropriate viewing
+// keys.
+func (s *SubscriptionManager) EncryptFilteredLogs(logs []*types.Log, rollupHash common.L2RootHash) (map[gethrpc.ID][]byte, error) {
```

Comment:
I'd maybe call it something like `EncryptSubscriptionLogs` or `EncryptAndFilterSubscriptionLogs` to be slightly clearer.